### PR TITLE
fix: Display transaction types in compact and detailed view pages [DEV-1895]

### PR DESCRIPTION
**Pull request by benyam7**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-1895 Fix UI bugs in block explorer](https://app.clickup.com/t/6600954/DEV-1895)

File: src/components/msg/unknown/index.tsx [hunk 1]
```diff
@@ -2,16 +2,18 @@ import React from 'react';
...
 import { useGetStyles } from './styles';
 
-const Unknown = (props: {
-  message: MsgUnknown;
+const Unknown = (props: { message: MsgUnknown;
 }) => {
   const { message } = props;
 
+  // our(cheqd) messages don't have "json" field
+  const data = message.json ?? message;
```
Comment (jay-dee7) [inline src/components/msg/unknown/index.tsx:10]: @benyam7 now we do have a `json` field?

File: src/components/msg/unknown/index.tsx [hunk 2]
```diff
@@ -2,16 +2,18 @@ import React from 'react';
...
+
   const { classes } = useGetStyles();
   return (
     <pre className={classes.root}>
       <code>
-        {JSON.stringify(message.json, null, '\t')}
```
Comment (jay-dee7) [inline src/components/msg/unknown/index.tsx:14]: if we do have a `json` field, then we can go back to `message.json`

File: src/components/msg/utils.tsx [hunk 3]
```diff
@@ -456,23 +474,31 @@ export const getMessageByType = (message: any, viewRaw: boolean, t:any) => {
...
+    ) {
       const log = R.pathOr(null, ['logs', i], transaction);
-      return model.fromJson(msg, log);
+      const data = model.fromJson(msg, log);
+      return data;
     }
-    return model.fromJson(msg);
+    const data = model.fromJson(msg);
+    // IDK why do we need to do this
+    data.type = msg['@type'];
```
Comment (jay-dee7) [inline src/components/msg/utils.tsx:500]: I think we should also have a `type` field so this shouldn't be needed?

**Review by jay-dee7 — CHANGES_REQUESTED** [review]
@benyam7 this PR looks good but we might need to check these comments out and see if everything works fine after resolving the comments

File: src/components/msg/utils.tsx (hunk 3 above)
Comment (benyam7) [reply src/components/msg/utils.tsx:500]: yep

Comment (benyam7) [reply src/components/msg/utils.tsx]: @jay-dee7 Okay, I made changes and tested them locally by re-building. It's working fine.

**Review by jay-dee7 — APPROVED** [review]
LGTM 💯

**Pull request merged**